### DISCUSSION TITLE
Question about the relationship between image quality and learned image codec

### DISCUSSION THREAD

**Discussion by achel-x** [tensorflow/compression#103] [Q&A] [answered]
In the hybrid codec, the QP(quantized parameter) can control the lost information from input image to the compressed image. 
So we can control the image quality by QP. I am confused **what is the key factor concerned with the image quality in learned image codec.** 
I mean if I want to adjust the image quality not by set \lambda to control the R-D trade-off, but maybe like QP or something else.
For example some setting in the network to control the different region quality( more important area has more bits and better quality, vice versa) under the same \lambda.

**Comment by jonaballe** [comment, marked as answer]
Conceptually, the main parameter to specify the rate–distortion tradeoff is lambda, even in hybrid video codecs.

In hybrid video codecs, the trade-off between rate and distortion is historically parameterized using the QP, which has been found to have a close-to monotonic relationship with lambda, as described in this paper: Sullivan, G.J. and Wiegand, T.: "Rate--distortion optimization for video compression", IEEE Signal Processing Magazine, 1998. So whether lambda or QP is given in hybrid codecs is basically equivalent.

In learned image compression models, we typically directly specify lambda to train a model that learns to solve the RD optimization problem in an amortized way (i.e. the encoder network learns how to solve the RD optimization non-iteratively for any given image, just by computing the network response, with the quantization step size held fixed, since the network can scale its output up or down as necessary).

**Reply by achel-x** [reply]
I got it. Thanks for your reply.

**Comment by ngcthuong** [comment]
Changing QP often can be equivalent to adjusting lambda. That's exactly many research in adaptive rate control doing.  One good thing is that there are several Lambda in the image/video codec, while QP is only one, which is easier to control the rate.

**Reply by achel-x** [reply]
Thanks for your insight. It helps me understand it.
